velox/functions/prestosql/types/TimestampWithTimeZoneType.cpp

```diff
@@ -53,15 +60,47 @@ void castFromTimestamp(
   });
 }
 
+void castFromString(
+    const SimpleVector<StringView>& inputVector,
+    exec::EvalCtx& context,
+    const SelectivityVector& rows,
+    VectorPtr& result) {
+  auto timestampWithTzVector = result->asFlatVector<int64_t>();
+  auto rawTimestampWithTzValues =
+      timestampWithTzVector->values()->asMutable<int64_t>();
+  timestampWithTzVector->clearNulls(rows);
+
+  context.applyToSelectedNoThrow(rows, [&](auto row) {
+    if (inputVector.isNullAt(row)) {
+      result->setNull(row, true);
+    } else {
+      auto tsTzStr = std::string_view(
+          inputVector.valueAt(row).data(), inputVector.valueAt(row).size());
+      auto [tsStr, tzStr] = splitTimestampWithTimezoneString(tsTzStr);
+      auto ts = util::fromTimestampString(tsStr);
+      auto tzID = util::getTimeZoneID(tzStr);
+      ts.toGMT(tzID);
+      int64_t outputMilliseconds = ts.getSeconds() * (int64_t)(1000);
```

Comment:
Instead of using the literal 1000 and the cast, you should use `kMillisecondsInSecond` from `DateTimeImpl.h` here.

Reply:
Done.